lib/Button/index.js

```diff
@@ -33,42 +36,31 @@ class Button extends Component {
     };
 
     this.handleOnClick = this.handleOnClick.bind(this);
+    this.getTypeClasses = this.getTypeClasses.bind(this);
   }
 
   handleOnClick() {
     this.setState({ disabled: this.props.disableOnClick });
     this.props.clickHandler();
   }
 
+  getTypeClasses() {
```

Comment:
"or to declare the component's usage for all cases in the actual style guide could also be a way"

I think that's a great idea. We'd be doing this anyways and using clear language in the storybook means it should be easy to understand how to use the types prop.

That way we resolve the issue with the switch statement and make it clear what types we support!